### PR TITLE
STYLE: f-strings

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Change to f-strings where possible.

Batch **1**

Addresses #2900.

### CODE REVIEW

**Comment by skoudoro** [comment]
Hi @DimitriPapadopoulos,

Do you an update concerning this PR? Thanks!

**Comment by DimitriPapadopoulos** [comment]
Also note that I have modified only a subset of the files. It will take some time and possibly multiple PRs to convert all files.

**Comment by skoudoro** [comment]
> Also note that I have modified only a subset of the files. It will take some time and possibly multiple PRs to convert all files.

make sense and I am fine with splitting in multiple PR.

File: dipy/reconst/mapmri.py [hunk 1]
```diff
@@ -267,21 +266,19 @@ def __init__(self,
                 raise ValueError('CVXPY package needed to enforce constraints.')
             if cvxpy_solver is not None:
                 if cvxpy_solver not in cvxpy.installed_solvers():
-                    msg = "Input `cvxpy_solver` was set to %s." % cvxpy_solver
-                    msg += " One of %s" % ', '.join(cvxpy.installed_solvers())
-                    msg += " was expected."
-                    raise ValueError(msg)
+                    installed_solvers = ', '.join(cvxpy.installed_solvers())
+                    raise ValueError(f"Input `cvxpy_solver` was set to"
```
Comment (skoudoro) [inline dipy/reconst/mapmri.py:270]: I do not think the f-string is necessary at this specific line. but I agree it is necessary below. 

There are others cases like that in your change, but that fine, I will go ahead and merge it. 

Thanks !

**Review by skoudoro — APPROVED** [review]
LGTM, Thanks @DimitriPapadopoulos 

merging this first batch.

File: dipy/reconst/mapmri.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply dipy/reconst/mapmri.py:270]: You're right, it's not really necessary. [Horizontally align quotes in multi-line f-string breaks](https://github.com/psf/black/issues/2915) is a good summary of how I ended up adding `f` to lines that do not really need it, without being overly happy with it.

Comment (skoudoro) [reply dipy/reconst/mapmri.py:270]: I see, thanks for the pointer!

**Pull request merged**